toolkits/landcover/impl/bands.js

```diff
@@ -146,9 +148,42 @@ function addFractionalYearBand(collection) {
   });
 }
 
+
+/**
+ * Compute a matrix multiply between the given image and a set of coefficients
+ * such as an eigenvector rotation or a Tasseled Cap transformation.
+ *
+ * The input image is expected to have the same number of bands as the width of
+ * the coefficients matrix.
+ *
+ * @param {!ee.Image} image The image on which to compute the transformation.
+ * @param {!Array<Array<number>>} coef The matrix to apply to the image.
+ * @param {?Array<string>} bandNames An optional list of output band names.
+ * @return {!ee.Image} The transformed bands.
+ */
+function computeMatrixMutliply(image, coef, bandNames) {
+  // Create some default band names if none were specified.
+  bandNames = bandNames || defaultBandNames("mmult", image.bandNames().length());
+
+  // make an Array Image with a 2-D Array per pixel.
+  var arrayImage = image.toArray().toArray(1);
+  return ee.Image(ee.Array(coef))
+      .matrixMultiply(arrayImage)
+      .arrayProject([0]) // get rid of the extra dimensions
+      .arrayFlatten([bandNames]);
+}
+
+/** Generate a series of band names of the form prefixN */
+function defaultBandNames(prefix, count) {
```

Comment:
My small programmer brain expects `defaultBandNames` to be a data structure with default band names, not an action (function). Any way to rename this so it's a verb, like `generateBandNames()`? I know "default" is also a verb, but that's not what I see when I read it.

Reply:
It's essentially private, so we can call it anything we want.

Done.